docs(guides): update basics tutorial series (#447)
EarthBuild/earthbuild: docs/basics/part-8a-using-earthly-in-your-current-ci.md

In this section, we will explore how to use EarthBuild in a CI system, such as GitHub Actions.

For more information on how to use EarthBuild in other CIs such as GitLab, Jenkins, or CircleCI, you can check out the [CI Integration page](../ci-integration/overview.md).

## Using EarthBuild in Your Current CI

To use EarthBuild in a CI, you typically encode the following steps in your CI's build configuration:

1. Download and install EarthBuild
2. Set up any credentials needed for the build
3. Log in to image registries, such as DockerHub
4. Run EarthBuild

> As part of this, you may need to set up credentials for Earthly Cloud, if you are using Earthly Satellites or Earthly Secrets. For this, you can use the following command:

Reviewer: ```suggestion
```

```bash
earthly account create-token my-ci-token

> ```

Reviewer: ```suggestion
```

Here is an explanation of the steps above:

> - The action `earthly/actions/setup-earthly@v1` downloads and installs EarthBuild. Running this action is similar to running the EarthBuild installation one-liner `sudo /bin/sh -c 'wget https://github.com/earthbuild/earthbuild/releases/download/v0.8.13/earthly-linux-amd64 -O /usr/local/bin/earthly && chmod +x /usr/local/bin/earthly'`

Reviewer: ```suggestion
- The action `earthbuild/actions-setup@v1` downloads and installs EarthBuild. Running this action is similar to running the EarthBuild installation one-liner `sudo /bin/sh -c 'wget https://github.com/earthbuild/earthbuild/releases/download/v0.8.13/earthly-linux-amd64 -O /usr/local/bin/earth && chmod +x /usr/local/bin/earth'`
```

- The command `docker login` performs a login to the DockerHub registry. This is required, to prevent rate-limiting issues when using popular base images.

> - The command `earthly --org ... --sat ... --ci --push +build` executes the build. The `--ci` flag is used here, in order to force the use of `--strict` mode. In `--strict` mode, EarthBuild prevents the use of features that make the build less repeatable and also disables local outputs -- because artifacts and images resulting from the build are not needed within the CI environment. Any outputs should be pushed via `RUN --push` or `SAVE IMAGE --push` commands. The flags `--org` and `--sat` allow you to select the organization and satellite to use for the build. If no satellite is specified, the build will be executed in the CI environment itself, with limited caching.

Reviewer: ```suggestion
- The command `earth --org ... --sat ... --ci --push +build` executes the build. The `--ci` flag is used here, in order to force the use of `--strict` mode. In `--strict` mode, EarthBuild prevents the use of features that make the build less repeatable and also disables local outputs -- because artifacts and images resulting from the build are not needed within the CI environment. Any outputs should be pushed via `RUN --push` or `SAVE IMAGE --push` commands. The flags `--org` and `--sat` allow you to select the organization and satellite to use for the build. If no satellite is specified, the build will be executed in the CI environment itself, with limited caching.
```
